File: gherkin/dart/lib/src/exceptions/UnexpectedEofException.dart
import 'package:gherkin/language.dart';

import 'TokenParserException.dart';

class UnexpectedEofException extends TokenParserException
{
  final String stateComment;

  final List<String> expectedTokenTypes;

  UnexpectedEofException(Token receivedToken, this.expectedTokenTypes, this.stateComment)
    : super(_getMessage(expectedTokenTypes), receivedToken);

  static String _getMessage(List<String> expectedTokenTypes)
    => 'unexpected end of file, expected: ${expectedTokenTypes.join(", ")}';

}

File: gherkin/php/src/ParserException/UnexpectedEofException.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin\ParserException;

use Cucumber\Gherkin\ParserException;
use Cucumber\Gherkin\Token;

final class UnexpectedEofException extends ParserException
{
    /**
     * @param non-empty-list<string> $expectedTokenTypes
     */
    public function __construct(
        public readonly Token $receivedToken,
        public readonly array $expectedTokenTypes,
        public readonly string $stateComment,
    ) {
        $message = sprintf(
            "unexpected end of file, expected: %s",
            join(', ', $expectedTokenTypes),
        );

        parent::__construct($message, $receivedToken->getLocation());
    }
}
